modules/data.atmosphere/R/download.NOAA_GEFS.R

```diff
@@ -0,0 +1,260 @@
+       humid_data[i] = PEcAn.data.atmosphere::rh2qair(humid_data[i], temperature_data[i], pressure_data[i])
+     }
+  } else {
+     for (i in 1:nrow(humid_data)) {
+       for (j in 1:ncol(humid_data)) {
+          humid_data[i,j] = PEcAn.data.atmosphere::rh2qair(humid_data[i,j], temperature_data[i,j], pressure_data[i,j])
+        }
+      }
+  }
+  
+  #Update the noaa_data list with the correct data
+  noaa_data[[humid_index]] <- humid_data
+  
+  # Convert NOAA's total precipitation (kg m-2) to precipitation flux (kg m-2 s-1)
+  #NOAA precipitation data is an accumulation over 6 hours.
+  precip_index = which(cf_var_names == "precipitation_flux")
+  
+  #The statement udunits2::ud.convert(1, "kg m-2 6 hr-1", "kg m-2 s-1") is equivalent to udunits2::ud.convert(1, "kg m-2 hr-1", "kg m-2 s-1") * 6,
+  #which is a little unintuitive. What will do the conversion we want is what's below:
+  noaa_data[[precip_index]] = udunits2::ud.convert(noaa_data[[precip_index]], "kg m-2 hr-1", "kg m-2 6 s-1")  #There are 21600 seconds in 6 hours
+  
+  #############################################
+  # Done with data processing.  Now writing the data to the specified directory. Each ensemble member is written to its own file, for a total
+  # of 21 files.  
+  if (!dir.exists(outfolder)) {
+    dir.create(outfolder, recursive=TRUE)
+  }
+  
+  # Create a data frame with information about the file.  This data frame's format is an internal PEcAn standard, and is stored in the BETY database to
+  # locate the data file.  The data file is stored on the local machine where the download occured.  Because NOAA GEFS is an 
+  # ensemble of 21 different forecast models, each model gets its own data frame.  All of the information is the same for 
+  # each file except for the file name.
+  results = data.frame(
+    file = "",                            #Path to the file (added in loop below).
+    host = PEcAn.remote::fqdn(),          #Name of the server where the file is stored
+    mimetype = "application/x-netcdf",    #Format the data is saved in
+    formatname = "CF Meteorology",        #Type of data
+    startdate = paste0(format(start_date, "%Y-%m-%dT%H:%M:00")),    #starting date and time, down to the second
+    enddate = paste0(format(end_date, "%Y-%m-%dT%H:%M:00")),        #ending date and time, down to the second
+    dbfile.name = "NOAA_GEFS",            #Source of data
+    stringsAsFactors = FALSE
+  )
+  
+  results_list = list()
+  
+  #Each ensemble gets its own file.
+  #These dimensions will be used for all 21 ncdf4 file members, so they're all declared once here.
+  #The data is really one-dimensional for each file (though we include lattitude and longitude dimensions
+  #to comply with the PEcAn standard).
+  time_dim = ncdf4::ncdim_def("Time", "6 Hours", 1:ncol(noaa_data[[1]]))
+  lat_dim = ncdf4::ncdim_def("Lattitude", "Degrees North", lat.in)
+  lon_dim = ncdf4::ncdim_def("Lattitude", "Degrees East", lon.in)
```

Comment:
Shouldn't this be `"Longitude"`? Also, "latitude" is misspelled (unless it's like that in the PEcAn standard?).

Reply:
Good catch.  I thing that was a copy and paste thing.  Fixed.